components/admin_console/group_settings/groups_list/index.ts

```diff
@@ -2,38 +2,50 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
-import {bindActionCreators} from 'redux';
+import {bindActionCreators, Dispatch, ActionCreatorsMapObject} from 'redux';
 import {createSelector} from 'reselect';
 
 import {linkLdapGroup, unlinkLdapGroup, getLdapGroups as fetchLdapGroups} from 'mattermost-redux/actions/admin';
 import {getLdapGroups, getLdapGroupsCount} from 'mattermost-redux/selectors/entities/admin';
 
-import GroupsList from './groups_list.jsx';
+import {GlobalState} from 'mattermost-redux/types/store';
+import {ActionFunc} from 'mattermost-redux/types/actions';
+import {GroupSearchOpts} from 'mattermost-redux/types/groups';
+
+import GroupsList from './groups_list';
+
+type Actions = {
+    getLdapGroups: (page?: number, perPage?: number, opts?: GroupSearchOpts) => Promise<{}>;
+    linkLdapGroup: (key: string) => Promise<{}>;
+    unlinkLdapGroup: (key: string) => Promise<{}>;
+}
 
 const getSortedListOfLdapGroups = createSelector(
     getLdapGroups,
     (ldapGroups) => {
-        const groups = Object.values(ldapGroups);
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const groups: any[] = Object.values(ldapGroups);
         groups.sort((a, b) => a.name.localeCompare(b.name));
         return groups;
     }
 );
 
-function mapStateToProps(state) {
+function mapStateToProps(state: GlobalState) {
     return {
         groups: getSortedListOfLdapGroups(state),
         total: getLdapGroupsCount(state),
     };
 }
 
-function mapDispatchToProps(dispatch) {
+function mapDispatchToProps(dispatch: Dispatch) {
     return {
-        actions: bindActionCreators({
+        actions: bindActionCreators<ActionCreatorsMapObject<ActionFunc>, Actions>({
             getLdapGroups: fetchLdapGroups,
             link: linkLdapGroup,
             unlink: unlinkLdapGroup,
         }, dispatch),
     };
 }
 
-export default connect(mapStateToProps, mapDispatchToProps)(GroupsList);
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export default connect(mapStateToProps, mapDispatchToProps)(GroupsList as any);
```

Comment:
This `as any` shouldn't be required if you're defining your types correctly, what's the error you're getting?